DDG4/edm4hep/Geant4Output2EDM4hep.cpp

```diff
@@ -460,14 +460,24 @@ void Geant4Output2EDM4hep::saveEvent(OutputContext<G4Event>& ctxt)  {
 
   // this does not compile as create() is we only get a const ref - need to review PODIO EventStore API
   edm4hep::EventHeaderCollection header_collection;
+
   auto header = header_collection.create();
   header.setRunNumber(runNumber);
   header.setEventNumber(eventNumber);
   header.setWeight(eventWeight);
   //not implemented in EDM4hep ?  header.setDetectorName(context()->detectorDescription().header().name());
-  header.setTimeStamp( std::time(nullptr) ) ;
-  m_frame.put( std::move(header_collection), "EventHeader");
+  header.setTimeStamp(std::time(nullptr));
+
+  // extract event header, in case we come from edm4hep input
+  auto* meh = context()->event().extension<edm4hep::MutableEventHeader>(false);
+  if(meh) {
+    header.setTimeStamp(meh->getTimeStamp());
```

Comment:
There is just the nullptr timestamp in line 469

Reply:
:+1: I misunderstood this whole thing here. We are copying over the contents from the input to the EventHeader that we are creating here. 

Just to be sure, I understood the rest correctly: The event & run number as well as the nominal event weight are "copied" by ingesting them in the reading side and then setting them again here on the output side.

Reply:
thanks for the explanation. Makes sense to me.

I think the EventHeader should not acquire much more information. It might be that we would need to follow such an approach with whatever we come up with for the generator info. But we can do that when we solve #1373